spec/middleware/metrics_spec.rb

```diff
@@ -9,13 +9,41 @@
     ::DiscoursePrometheus::Middleware::Metrics.new(app)
   end
 
+  it "will 404 for unauthed if prometheus_trusted_ip_whitelist_regex is unset" do
+    status, = middleware.call("PATH_INFO" => '/metrics', "REMOTE_ADDR" => '200.0.1.1', "rack.input" => StringIO.new)
+    expect(status).to eq(404)
+  end
+
   it "will 404 for unauthed" do
     status, = middleware.call("PATH_INFO" => '/metrics', "REMOTE_ADDR" => '200.0.1.1', "rack.input" => StringIO.new)
     expect(status).to eq(404)
   end
 
+  it "will 404 for unauthed and invalid regex" do
+    GlobalSetting.stubs(:prometheus_trusted_ip_whitelist_regex).returns("unbalanced bracket[")
```

Comment:
no need to stub just use this pattern: 

https://github.com/discourse/discourse/blob/b8d2261db9f216414b86a27d83435f4ee99561cc/spec/models/global_setting_spec.rb#L20-L20